TestBenches/TrackQuality_test.cpp

```diff
@@ -0,0 +1,78 @@
+//Track Quality test bench 
+
+#include "TrackQualityTop.h"
+#include "TrackQualityMemory.h"
+#include "parameters.h"
+#include "FileReadUtility.h"
+#include "BDT.h"
+
+#include <fstream>
+#include <iostream>
+#include <algorithm>
+#include <vector>
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+
+const int nevents = 5;  // number of events to run
+using namespace std;
+
+int main()
+{
+  //load input data from text file
+  ifstream fin_tracks;
+  bool valid_intrack = openDataFile(fin_tracks,"TQ/hls_hex.dat");
+  if (not valid_intrack) return -1;
+  //load predictions from text file
+  ifstream fin_predictions;
+  bool valid_inpred = openDataFile(fin_predictions,"TQ/conifer_predictions.dat");
+  if (not valid_inpred) return -1;
+
+  std::string trk_line;
+  std::string pred_line;
+  int err_count = 0;
+  int event_count = 0;
+
+  while (event_count < nevents) {
+    
+    std::getline(fin_tracks,trk_line);
+    
+    if (trk_line.find("Event") != std::string::npos){
+      event_count ++;
+      continue;
+    }
+    else{
+      std::getline(fin_predictions,pred_line); //predicitions don't have event headers so only iterate when input data is tracks
+
+      char pDelimeter = '|';
+      std::vector<std::string> splitline = split(trk_line, pDelimeter);
+      
+      ap_uint<96> x = 0;
```

Comment:
Could you use TTTrack to get rid of the magic number 96 here? You could at least do
```c++
ap_uint<TTTrack::getWidth()> x = 0;
```
or you could even add a constructor to TTTrack that takes in a string and does the conversion that you do here. Then you could just call it as
```c++
TTTrack inputTrack(splitline[0]);
```